heim-process/src/sys/macos/bindings/process.rs

```diff
@@ -202,18 +208,25 @@ pub fn processes() -> Result<Vec<kinfo_proc>, Error> {
                 0,
             )
         };
-        match result {
-            libc::ENOMEM => continue,
-            code if code < 0 => return Err(Error::last_os_error().with_sysctl(name.as_ref())),
-            _ => {
-                let length = size / mem::size_of::<kinfo_proc>();
-                unsafe {
-                    processes.set_len(length);
-                }
-                debug_assert!(!processes.is_empty());
-
-                return Ok(processes);
+
+        if result < 0 {
+            // `libc::ENOMEM` indicates there was not enough space in `processes` to store the whole
+            // process list which can occur when a new process spawns between getting the size and
+            // storing. If this is the case then simply try again.
+            if io::Error::last_os_error().raw_os_error().unwrap() == libc::ENOMEM {
```

Comment:
I don't think there a possibility to get any other `Error` between `libc::sysctl()` and `io::Error::last_os_error()` calls, but any `.unwrap` generally should be avoided, usual matching should be used instead:

```rust
            let e = io::Error::last_os_error();
            if let Some(libc::ENOMEM) == e.raw_os_error() {
                continue;
            } else {
                return Err(e.with_sysctl(name.as_ref()));
            }
```

Reply:
That's fair. `None` should never return for `raw_os_error` constructed from `last_os_error`, but it's bad to get in the habit of relying on things like this. I'll update with the changes and push when we decide on the action for the other comment.